src/main/java/com/synopsys/integration/alert/provider/polaris/PolarisApiHelper.java

```diff
@@ -0,0 +1,162 @@
+import com.synopsys.integration.polaris.common.service.IssueService;
+import com.synopsys.integration.polaris.common.service.RoleAssignmentsService;
+import com.synopsys.integration.polaris.common.service.UserService;
+
+public class PolarisApiHelper {
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private final IssueService issueService;
+    private final RoleAssignmentsService roleAssignmentsService;
+    private final UserService userService;
+
+    public PolarisApiHelper(final IssueService issueService, final RoleAssignmentsService roleAssignmentsService, final UserService userService) {
+        this.issueService = issueService;
+        this.roleAssignmentsService = roleAssignmentsService;
+        this.userService = userService;
+    }
+
+    public Set<PolarisIssueModel> createIssueModelsForProject(final String projectId, final String projectName, final List<String> branchIds) {
+        final Set<PolarisIssueModel> issuesForProjectFromServer = new HashSet<>();
+        for (final String branchId : branchIds) {
+            try {
+                final List<QueryIssueResource> foundIssues = issueService.getIssuesForProjectAndBranch(projectId, branchId);
+                final Map<String, Integer> issueTypeCounts = mapIssueTypeToCount(foundIssues);
+
+                for (final Map.Entry<String, Integer> issueTypeEntry : issueTypeCounts.entrySet()) {
+                    final PolarisIssueModel newIssue = PolarisIssueModel.createNewIssue(issueTypeEntry.getKey(), issueTypeEntry.getValue());
+                    issuesForProjectFromServer.add(newIssue);
+                }
+            } catch (final IntegrationException e) {
+                logger.error("Problem getting issues from Polaris: {}", projectName, e);
+                continue;
+            }
+        }
+        return issuesForProjectFromServer;
+    }
+
+    public List<String> getBranchesIdsForProject(final List<BranchV0Resource> branches) {
+        return branches
+                   .stream()
+                   .map(BranchV0Resource::getId)
+                   .collect(Collectors.toList());
+    }
+
+    public Set<String> getAllEmailsForProject(final ProjectV0Resource project) throws IntegrationException {
+        List<UserResource> allUsers = null;
+        final RoleAssignmentResources projectRoleAssignements = roleAssignmentsService.getRoleAssignmentsForProjectWithIncluded(project.getId(),
+            RoleAssignmentsService.INCLUDE_USERS, RoleAssignmentsService.INCLUDE_GROUPS);
+
+        final Set<String> emails = new HashSet<>();
+        for (final RoleAssignmentResource roleAssignment : projectRoleAssignements.getData()) {
+            final Optional<String> optionalUserEmail = getEmailForRoleAssignedUser(projectRoleAssignements, roleAssignment);
+            if (optionalUserEmail.isPresent()) {
+                emails.add(optionalUserEmail.get());
+            } else {
+                if (null == allUsers) {
```

Comment:
If there is no role assignment, we assume all Users are assigned to this project?

Reply:
No, it's just lazily initializing the local variable.